staging/src/k8s.io/api/discovery/v1alpha1/types.go

```diff
@@ -59,21 +59,26 @@ type EndpointSlice struct {
 type AddressType string
 
 const (
-	// AddressTypeIP represents an IP Address. Inclusive of IPv4 and IPv6
-	// addresses.
+	// AddressTypeIP represents an IP Address.
+	// This address type has been deprecated and has been replaced by the IPv4
+	// and IPv6 adddress types. New resources with this address type will be
+	// considered invalid. This will be fully removed in 1.18.
+	// +deprecated
 	AddressTypeIP = AddressType("IP")
-	// AddressTypeFQDN represents a Fully Qualified Domain Name.
+	// AddressTypeIPv4 represents an IPv4 Address.
+	AddressTypeIPv4 = AddressType(v1.IPv4Protocol)
+	// AddressTypeIPv6 represents an IPv6 Address.
+	AddressTypeIPv6 = AddressType(v1.IPv6Protocol)
+	// AddressTypeFQDN represents a FQDN.
 	AddressTypeFQDN = AddressType("FQDN")
 )
 
 // Endpoint represents a single logical "backend" implementing a service.
 type Endpoint struct {
 	// addresses of this endpoint. The contents of this field are interpreted
-	// according to the corresponding EndpointSlice addressType field. This
-	// allows for cases like dual-stack networking where both IPv4 and IPv6
-	// addresses would be included with the IP addressType. Consumers (e.g.
-	// kube-proxy) must handle different types of addresses in the context of
-	// their own capabilities. This must contain at least one address but no
+	// according to the corresponding EndpointSlice addressType field. Consumers
+	// (e.g. kube-proxy) must handle different types of addresses in the context
```

Comment:
We have bigger dreams for EndpointSlice (I remember possible mesh usecase). right? if so then let us remove reference to kube-proxy. It might limit the imagination of the reader.

Reply:
Good call - removed the kube-proxy reference.